test/vidom/runtime-test.js

```diff
@@ -0,0 +1,1208 @@
+var assert = require('assert');
+var fixtures = require('../fixtures')('vidom');
+var bemxjst = require('../../').vidom;
+
+var test = fixtures.test;
+
+describe('VIDOM compiler/Runtime', function() {
+
+  // FIXME modify this test to check cases with
+  // https://ru.bem.info/libs/bem-components/v2.4.0/desktop/radio-group/
+  it('should render many items in content' , function() {
+    test(function() {
+      block('b1')(
+        content()(function() {
+          return [
+            { block: 'b2' },
+            { block: 'b3' }
+          ];
+        })
+      )
+    }, { block: 'b1' },
+      [ 'div', { className: 'b1' },
+        [ 'div', { className: 'b2' } ], [ 'div', { className: 'b3' } ] ]);
+  });
+
+  it('should not render ReactElement', function() {
+    test(function() {},
+      {
+        block: 'b1',
+        content: [
+          { $$typeof: 'xxx' },
+          { block: 'b2' },
+          { $$typeof: 'yyy' }
+        ]
+      }, [ 'div', { className: 'b1' },
+          { $$typeof: 'xxx' },
+          [ 'div', { className: 'b2' } ],
+          { $$typeof: 'yyy' } ]);
+  });
+
+  it('should render content as array', function() {
+    test(function() {},
+      { tag: 'section', content: [
+        { block: 'b1' },
+        { block: 'b2' }
+      ] }, [ 'section', null,
+        [ 'div', { className: 'b1' } ],
+        [ 'div', { className: 'b2' } ] ]);
+  });
+
+  xit('should apply("content")', function() {
+    test(function() {
+      block('b1').def()(function() {
+        return applyCtx([ apply('content'), this.ctx.tag ])
+      });
+    }, { block: 'b1', tag: '1', content: 'hahaha' },
+      [ 'div', null, [ 'span', null, 'hahaha' ], [ 'span', null, '1' ] ]);
+  });
+
+  it('should render children with different types', function() {
+    test(function() {
+      block('b1')(
+        content()(function() {
+          return [
+            's1',
+            { elem: 'e1', content: 's' },
+            's2',
+            { elem: 'e2', content: 's' }
+           ];
+        })
+      );
+    },
+    { block: 'b1' },
+    [ 'div', { className: 'b1' },
+      's1',
+      [ 'div', { className: 'b1__e1' }, 's' ],
+      's2',
+      [ 'div', { className: 'b1__e2' }, 's' ] ]
+    )
+  });
+
+  it('should wrap text into container', function() {
+    test(function() {},
+      [ 'str1', 'str2' ],
+      [ 'div', null, [ 'span', null, 'str1' ], [ 'span', null, 'str2' ] ]);
```

Comment:
but maybe it's a good idea to return [ 'span', null, 'str1', 'str2'  ]
We'll try to fix it